Apps/WebClient/src/ClientApp/src/views/timeline.vue

```diff
@@ -346,347 +308,25 @@ export default class TimelineView extends Vue {
-            .finally(() => {
-                this.isCommentLoading = false;
-            });
-    }
-
-    private loadImmunizationEntries() {
-        if (this.immunizationLoadReady) {
-            this.immunizationLoadDeferred = false;
-            this.immunizationLoadReady = false;
-        }
-        // Add the immunization entries to the timeline list
-        for (let immunization of this.patientImmunizations) {
-            this.timelineEntries.push(
-                new ImmunizationTimelineEntry(immunization)
-            );
-        }
-        this.sortEntries();
-        this.setFilterTypeCount(
-            EntryType.Immunization,
-            this.patientImmunizations.length
-        );
-    }
-
-    private onEntryAdded(entry: TimelineEntry) {
-        this.logger.debug(`Timeline Entry added: ${JSON.stringify(entry)}`);
-        this.isAddingNote = false;
-        if (entry) {
-            this.timelineEntries.push(entry);
-            this.sortEntries();
-            if (entry.type === EntryType.Note) {
-                this.addFilterTypeCount(EntryType.Note, 1);
-            }
-        }
-    }
-
-    private onEntryEdit() {
-        this.isEditingEntry = true;
-    }
-
-    private onNoteEditClose() {
-        this.isEditingEntry = false;
-        this.isAddingNote = false;
-    }
-
-    private onEntryUpdated(entry: TimelineEntry) {
-        this.logger.debug(`Timeline Entry updated: ${JSON.stringify(entry)}`);
-        const index = this.timelineEntries.findIndex(
-            (e) => e.id === entry.id && e.type === entry.type
-        );
-        this.timelineEntries.splice(index, 1);
-        this.timelineEntries.push(entry);
-        this.isEditingEntry = false;
-        this.sortEntries();
-    }
-
-    private onEntryDeleted(entry: TimelineEntry) {
-        this.logger.debug(`Timeline Entry deleted: ${JSON.stringify(entry)}`);
-        const index = this.timelineEntries.findIndex((e) => e.id == entry.id);
-        this.timelineEntries.splice(index, 1);
-        this.sortEntries();
-        if (entry.type === EntryType.Note) {
-            this.addFilterTypeCount(EntryType.Note, -1);
-        }
-    }
-
-    private onProtectiveWordSubmit(value: string) {
-        this.fetchMedicationStatements(value);
-    }
-
-    private onProtectiveWordCancel() {
-        // Does nothing as it won't be able to fetch pharmanet data.
-        this.logger.debug("protective word cancelled");
+    private fetchTimelineData() {
+        Promise.all([
+            this.retrieveMedications({ hdid: this.user.hdid }),
+            this.retrieveImmunizations({ hdid: this.user.hdid }),
+            this.retrieveLaboratory({ hdid: this.user.hdid }),
+            this.retrieveEncounters({ hdid: this.user.hdid }),
+            this.retrieveNotes({ hdid: this.user.hdid }),
+            this.retrieveComments({ hdid: this.user.hdid }),
+        ]).catch((err) => {
+            this.logger.error(`Error loading timeline data: ${err}`);
+        });
```

Comment:
All data is retrieved and stored in VUEX if it was not there before. That way there is no need to handle the state of the retrieval in the timeline